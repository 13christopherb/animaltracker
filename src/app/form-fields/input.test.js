import React from 'react';
import {mount} from 'enzyme';
import {Input} from './';
import renderer from 'react-test-renderer'

describe('<Input />', () => {
    const defaultProps = {
        meta: {
            touched: false,
            error: null
        },
        id: 'name',
        name: 'name',
        title: 'Name',
        type: 'text',
    };
    describe('render()', () => {
        test('should render the component normally', () => {
            const component = renderer.create(<Input {...defaultProps} />).toJSON();
            expect(component).toMatchSnapshot()
        });
        test('should highlight incorrect field', () => {
            const props = {
                ...defaultProps,
                meta: {
                    touched: true,
                    error: 'Required'
                }
            };
            const field = mount(<Input {...props} />).find('input');
            expect(field.hasClass('is-invalid')).toEqual(true);
        });
        test('should display error message', () => {
            const props = {
                ...defaultProps,
                meta: {
                    touched: true,
                    error: 'Required'
                }
            };
            const feedback = mount(<Input {...props} />).find('.invalid-feedback');
            expect(feedback.text()).toEqual(props.meta.error);
        })
    });
});