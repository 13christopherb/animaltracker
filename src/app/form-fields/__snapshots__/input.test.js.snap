// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> render() should render the component normally 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label form-label"
    htmlFor="name"
  >
    Name
  </label>
  <div
    className="input-group"
  >
    <input
      className="form-control"
      id="name"
      name="name"
      type="text"
    />
    <div
      className="invalid-feedback"
    />
  </div>
</div>
`;
